### PR TITLE
Update logging docs

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

Part of #7666

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @carmocca @edward-io @ananthsub

### CODE REVIEW

File: docs/source/extensions/logging.rst [hunk 1]
```diff
@@ -75,55 +108,79 @@ Lightning offers automatic log functionalities for logging scalars, or manual lo
...
+   * - Hook
+     - on_step
+     - on_epoch
+   * - on_before_backward, on_after_backward, on_before_optimizer_step, on_before_zero_grad
+     - False
+     - True
+   * - on_train_start, on_train_epoch_start, on_train_epoch_end, training_epoch_end
+     - False
+     - True
+   * - on_train_batch_start, on_train_batch_end
```
Comment (rohitgr7) [inline docs/source/extensions/logging.rst:153]: don't resolve this one.. need to update once https://github.com/PyTorchLightning/pytorch-lightning/pull/10756 gets merged.

**Comment by rohitgr7** [comment]
here are the docs: https://140072-178626720-gh.circle-artifacts.com/0/html/extensions/logging.html

File: docs/source/extensions/logging.rst [hunk 2]
```diff
@@ -14,45 +14,78 @@
...
+Supported Loggers
+=================
+
+The following are loggers we support:
+
+.. note::
+    The following loggers will normally plot an additional chart (**global_step VS epoch**).
+
+.. note::
+    Depending on the loggers you use, there might be some additional charts.
```
Comment (awaelchli) [inline docs/source/extensions/logging.rst:26]: Can we
1) move the notes to the bottom (seems distracting especially directly in the introduction of the main section here
2) combine the two into one note (could be bullet points)

Alternatively just make it regular text (which I would prefer), because these are items that are anyway immediately visible once they open the logger.

File: docs/source/extensions/logging.rst [hunk 3]
```diff
@@ -75,55 +108,79 @@ Lightning offers automatic log functionalities for logging scalars, or manual lo
...
 
 
-If your work requires to log in an unsupported function, please open an issue with a clear description of why it is blocking you.
+If your work requires to log in an unsupported method, please open an issue with a clear description of why it is blocking you.
 
 
-Manual logging
-==============
-If you want to log anything that is not a scalar, like histograms, text, images, etc... you may need to use the logger object directly.
+Manual logging Non-Scalar Artifacts
```
Comment (awaelchli) [inline docs/source/extensions/logging.rst:181]: I strongly suggest to either make all titles properly capitalized or not at all. For consistency. For the former, when in doubt, I recommend https://capitalizemytitle.com/

**Review by awaelchli — APPROVED** [review]

File: docs/source/extensions/logging.rst [hunk 4]
```diff
@@ -217,27 +265,26 @@ Logging frequency
...
    trainer = Trainer(log_every_n_steps=k)
 
 
-
 Log writing frequency
 =====================
 
 Writing to a logger can be expensive, so by default Lightning writes logs to disk or to the given logger every 100 training steps.
-To change this behaviour, set the interval at which you wish to flush logs to the filesystem using the `flush_logs_every_n_steps` :class:`~pytorch_lightning.trainer.trainer.Trainer` flag.
+To change this behaviour, set the interval at which you wish to flush logs to the filesystem using the ``flush_logs_every_n_steps`` :class:`~pytorch_lightning.trainer.trainer.Trainer` flag.
```
Comment (tchaton) [inline docs/source/extensions/logging.rst:280]:
```suggestion
If you wish to flush logs to the filesystem at a different steps rate, use the ``flush_logs_every_n_steps`` :class:`~pytorch_lightning.trainer.trainer.Trainer` flag.
```

File: pytorch_lightning/core/lightning.py [hunk 5]
```diff
@@ -312,18 +312,7 @@ def log(
...
             self.log('train_loss', loss)
 
-        The default behavior per hook is as follows:
-
-        .. csv-table:: ``*`` also applies to the test loop
-           :header: "LightningModule Hook", "on_step", "on_epoch", "prog_bar", "logger"
-           :widths: 20, 10, 10, 10, 10
-
-           "training_step", "T", "F", "F", "T"
-           "training_step_end", "T", "F", "F", "T"
```
Comment (tchaton) [inline pytorch_lightning/core/lightning.py:322]: I believe this one was quite useful. Helped me to understand the default quite fast.

Comment (rohitgr7) [reply pytorch_lightning/core/lightning.py:322]: This is added to the docs now. Because we have more hooks that are not mentioned here, so I moved them inside docs and linked them to the API.

File: docs/source/extensions/logging.rst (hunk 3 above)
Comment (rohitgr7) [reply docs/source/extensions/logging.rst]: thanks! updated!!

File: docs/source/extensions/logging.rst (hunk 2 above)
Comment (rohitgr7) [reply docs/source/extensions/logging.rst]: done!

File: pytorch_lightning/core/lightning.py (hunk 5 above)
Comment (rohitgr7) [reply pytorch_lightning/core/lightning.py:322]: here: https://140072-178626720-gh.circle-artifacts.com/0/html/extensions/logging.html#id4

File: docs/source/extensions/logging.rst [hunk 6]
```diff
@@ -14,56 +14,85 @@
...
+    # Install tensorboard
+    pip install tensorboard
     tensorboard --logdir=lightning_logs/
 
+To run tensorboard in a jupyter notebook environment, use the following in a jupyter cell:
+
+.. code-block:: bash
+
+    %reload_ext tensorboard
+    %tensorboard --logdir=lightning_logs/
```
Comment (tchaton) [inline docs/source/extensions/logging.rst:62]: Neat !

File: docs/source/extensions/logging.rst [hunk 7]
```diff
@@ -14,56 +14,85 @@
...
 .. note::
 
-    By default, lightning logs every 50 steps. Use Trainer flags to :ref:`logging_frequency`.
+    By default, Lightning logs every 50 steps. Use Trainer flags to :ref:`logging_frequency`.
 
 .. note::
 
-    All loggers log by default to `os.getcwd()`. To change the path without creating a logger set
-    `Trainer(default_root_dir='/your/path/to/save/checkpoints')`
+    All loggers log by default to ``os.getcwd()``. To change the path without creating a logger set
```
Comment (tchaton) [inline docs/source/extensions/logging.rst:94]:
```suggestion
    By default, all loggers log to ``os.getcwd()``. You can change the log path by doing ``Trainer(default_root_dir='/your/path/to/save/checkpoints')`` without instantiating a logger.
```

File: docs/source/extensions/logging.rst [hunk 8]
```diff
@@ -75,55 +104,76 @@ Lightning offers automatic log functionalities for logging scalars, or manual lo
...
 .. code-block:: python
 
     def training_step(self, batch, batch_idx):
         self.log("my_metric", x)
 
 
-    # or a dict
+    # or a dict to get multiple metrics on the same plot of the logger supports it
     def training_step(self, batch, batch_idx):
         self.log("performance", {"acc": acc, "recall": recall})
```
Comment (tchaton) [inline docs/source/extensions/logging.rst:118]: Or use:

```py
self.log("performance/acc", acc)
self.log("performance/recall", recall)
```

File: docs/source/extensions/logging.rst [hunk 9]
```diff
@@ -75,55 +104,76 @@ Lightning offers automatic log functionalities for logging scalars, or manual lo
...
 
-Depending on where log is called from, Lightning auto-determines the correct logging mode for you. \
-But of course you can override the default behavior by manually setting the :func:`~~pytorch_lightning.core.lightning.LightningModule.log` parameters.
+Depending on where log is called from, Lightning auto-determines the correct logging mode for you. But of course you can
```
Comment (tchaton) [inline docs/source/extensions/logging.rst:120]:
```suggestion
Depending on where :meth:`~pytorch_lightning.core.lightning.LightningModule.log` method is called, Lightning auto-determines the correct logging mode for you. Of course you can
```

File: docs/source/extensions/logging.rst [hunk 10]
```diff
@@ -75,55 +104,76 @@ Lightning offers automatic log functionalities for logging scalars, or manual lo
...
-* `prog_bar`: Logs to the progress bar.
-
-* `logger`: Logs to the logger like Tensorboard, or any other custom logger passed to the :class:`~pytorch_lightning.trainer.trainer.Trainer`.
-
+The :meth:`~pytorch_lightning.core.lightning.LightningModule.log` method has a few options:
+
+* ``on_step``: Logs the metric at the current step.
+* ``on_epoch``: Automatically accumulates and logs at the end of the epoch.
+* ``prog_bar``: Logs to the progress bar (Default: ``False``).
+* ``logger``: Logs to the logger like ``Tensorboard``, or any other custom logger passed to the
```
Comment (tchaton) [inline docs/source/extensions/logging.rst:133]: Let's list them all. `sync_dist`, etc..

**Review by tchaton — APPROVED** [review]
LGTM ! After comments are resolved.

File: docs/source/extensions/logging.rst [hunk 11]
```diff
@@ -75,55 +104,76 @@ Lightning offers automatic log functionalities for logging scalars, or manual lo
...
+method to log from anywhere in a :doc:`lightning module <../common/lightning_module>` and :doc:`callbacks <../extensions/callbacks>`.
 
 .. code-block:: python
 
     def training_step(self, batch, batch_idx):
         self.log("my_metric", x)
 
 
-    # or a dict
+    # or a dict to get multiple metrics on the same plot of the logger supports it
```
Comment (rohitgr7) [inline docs/source/extensions/logging.rst:116]:
```suggestion
    # or a dict to get multiple metrics on the same plot if the logger supports it
```

File: docs/source/extensions/logging.rst (hunk 8 above)
Comment (rohitgr7) [reply docs/source/extensions/logging.rst:118]: no.. this is different. Your example will create `performance` and will create separate plots for each for acc and recall. The existing example will plot recall and acc on the same plot titled `performance`.

File: docs/source/extensions/logging.rst [hunk 12]
```diff
@@ -14,56 +14,85 @@
 Logging
 #######
 
-Lightning supports the most popular logging frameworks (TensorBoard, Comet, etc...).
```
Comment (kaushikb11) [inline docs/source/extensions/logging.rst:17]: I liked the statement as it emphasis on our good support for loggers

File: docs/source/extensions/logging.rst [hunk 13]
```diff
@@ -14,56 +14,85 @@
...
+    NeptuneLogger
+    TensorBoardLogger
+    TestTubeLogger
+    WandbLogger
 
-By default, Lightning uses `PyTorch TensorBoard <https://pytorch.org/docs/stable/tensorboard.html>`__ logging  under the hood, and stores the logs to a directory (by default in ``lightning_logs/``).
+
+The above loggers will normally plot an additional chart (**global_step VS epoch**). Depending on the loggers you use, there might be some additional charts too.
+
+By default, Lightning uses ``TensorBoard`` logger under the hood, and stores the logs to a directory (by default in ``lightning_logs/``).
```
Comment (kaushikb11) [inline docs/source/extensions/logging.rst:40]: Could you hyperlink PyTorch Tensorboard?

File: docs/source/extensions/logging.rst [hunk 14]
```diff
@@ -14,56 +14,85 @@
...
 
 .. testcode::
 
     from pytorch_lightning import Trainer
 
-    # Automatically logs to a directory
-    # (by default ``lightning_logs/``)
+    # Automatically logs to a directory (by default lightning_logs/)
```
Comment (kaushikb11) [inline docs/source/extensions/logging.rst:46]:
```suggestion
    # Automatically logs to a directory (by default ``lightning_logs/``)
```

File: docs/source/extensions/logging.rst [hunk 15]
```diff
@@ -14,56 +14,85 @@
...
     trainer = Trainer()
 
 To see your logs:
 
 .. code-block:: bash
 
+    # Install tensorboard
+    pip install tensorboard
```
Comment (kaushikb11) [inline docs/source/extensions/logging.rst:54]: Isn't it installed by default for Lightning? I don't think the installment command is necessary

File: docs/source/extensions/logging.rst [hunk 16]
```diff
@@ -14,56 +14,85 @@
...
     tensorboard --logdir=lightning_logs/
 
+To run tensorboard in a jupyter notebook environment, use the following in a jupyter cell:
```
Comment (kaushikb11) [inline docs/source/extensions/logging.rst:57]:
```suggestion
To visualize tensorboard in a jupyter notebook environment, run the following command in a jupyter cell:
```

File: docs/source/extensions/logging.rst [hunk 17]
```diff
@@ -14,56 +14,85 @@
...
 .. note::
 
-    By default, lightning logs every 50 steps. Use Trainer flags to :ref:`logging_frequency`.
+    By default, Lightning logs every 50 steps. Use Trainer flags to :ref:`logging_frequency`.
 
 .. note::
 
-    All loggers log by default to `os.getcwd()`. To change the path without creating a logger set
-    `Trainer(default_root_dir='/your/path/to/save/checkpoints')`
+    By default, all loggers log to ``os.getcwd()``. You can change the log path using
```
Comment (kaushikb11) [inline docs/source/extensions/logging.rst:94]:
```suggestion
    By default, all loggers log to ``os.getcwd()``. You can change the logging path using
```

File: docs/source/extensions/logging.rst [hunk 18]
```diff
@@ -75,55 +104,88 @@ Lightning offers automatic log functionalities for logging scalars, or manual lo
...
 
 
-    # or a dict
+    # or a dict to get multiple metrics on the same plot if the logger supports it
     def training_step(self, batch, batch_idx):
         self.log("performance", {"acc": acc, "recall": recall})
 
-Depending on where log is called from, Lightning auto-determines the correct logging mode for you. \
-But of course you can override the default behavior by manually setting the :func:`~~pytorch_lightning.core.lightning.LightningModule.log` parameters.
+Depending on where Use the :meth:`~pytorch_lightning.core.lightning.LightningModule.log` method is called, Lightning auto-determines
```
Comment (kaushikb11) [inline docs/source/extensions/logging.rst:120]:
```suggestion
Depending on where the :meth:`~pytorch_lightning.core.lightning.LightningModule.log` method is called, Lightning auto-determines
```

File: docs/source/extensions/logging.rst [hunk 19]
```diff
@@ -75,55 +104,88 @@ Lightning offers automatic log functionalities for logging scalars, or manual lo
...
-If your work requires to log in an unsupported function, please open an issue with a clear description of why it is blocking you.
+If your work requires to log in an unsupported method, please open an issue with a clear description of why it is blocking you.
 
 
-Manual logging
-==============
-If you want to log anything that is not a scalar, like histograms, text, images, etc... you may need to use the logger object directly.
+Manual Logging Non-Scalar Artifacts
+===================================
+If you want to log anything that is not a scalar, like histograms, text, images, etc. you may need to use the logger object directly.
```
Comment (kaushikb11) [inline docs/source/extensions/logging.rst:188]:
```suggestion
If you want to log anything that is not a scalar, like histograms, text, images, etc., you may need to use the logger object directly.
```

File: docs/source/extensions/logging.rst (hunk 12 above)
Comment (rohitgr7) [reply docs/source/extensions/logging.rst:17]: we have them right below: https://140783-178626720-gh.circle-artifacts.com/0/html/extensions/logging.html

File: docs/source/extensions/logging.rst (hunk 13 above)
Comment (rohitgr7) [reply docs/source/extensions/logging.rst:40]: we have already listed with links 2 lines above: https://140783-178626720-gh.circle-artifacts.com/0/html/extensions/logging.html

File: docs/source/extensions/logging.rst (hunk 15 above)
Comment (rohitgr7) [inline docs/source/extensions/logging.rst:54]:
```suggestion
```

**Review by kaushikb11 — APPROVED** [review]

File: docs/source/extensions/logging.rst [hunk 20]
```diff
@@ -14,16 +14,36 @@
...
+    :toctree: generated
+    :nosignatures:
+    :template: classtemplate.rst
+
+    CometLogger
+    CSVLogger
+    MLFlowLogger
+    NeptuneLogger
+    TensorBoardLogger
+    TestTubeLogger
```
Comment (carmocca) [inline docs/source/extensions/logging.rst:34]: Should we avoid mentioning `TestTubeLogger`? It's deprecated and unmaintained.

File: docs/source/extensions/logging.rst [hunk 21]
```diff
@@ -300,19 +354,19 @@ Read more about custom Python logging `here <https://docs.python.org/3/library/l
...
+When Lightning creates a checkpoint, it stores a key ``"hyper_parameters"`` with the hyperparams.
 
 .. code-block:: python
 
     lightning_checkpoint = torch.load(filepath, map_location=lambda storage, loc: storage)
     hyperparams = lightning_checkpoint["hyper_parameters"]
 
 Some loggers also allow logging the hyperparams used in the experiment. For instance,
-when using the TestTubeLogger or the TensorBoardLogger, all hyperparams will show
+when using the ``TestTubeLogger`` or the ``TensorBoardLogger``, all hyperparams will show
```
Comment (carmocca) [inline docs/source/extensions/logging.rst:369]: Same comment about `TestTubeLogger`

**Review by carmocca — APPROVED** [review]

File: docs/source/extensions/logging.rst (hunk 20 above)
Comment (rohitgr7) [reply docs/source/extensions/logging.rst:34]: yeah, and it will be removed within v1.7? docs will be updated too in the removal PR.

Comment (carmocca) [reply docs/source/extensions/logging.rst:34]: I am personally not a fan of having deprecated features in our docs at all. As an user, I'd rather not learn about this only to later notice it will be removed.

But if you prefer leaving this here until it's finally removed, that's okay. you've got my approval :P

Comment (kaushikb11) [reply docs/source/extensions/logging.rst:34]: I agree with @carmocca, we should remove it.

Comment (rohitgr7) [reply docs/source/extensions/logging.rst:34]: yeah makes sense.. will update :)

**Review by Borda — APPROVED** [review]
